tests/run-with-compiler/i5144.scala

```diff
@@ -0,0 +1,17 @@
+import scala.quoted._
+
+object Test {
+  implicit val toolbox: scala.quoted.Toolbox = scala.quoted.Toolbox.make
+
+  def eval1(ff: Expr[Int => Int]): Expr[Int] = '((~ff)(42))
+
+  def peval1(): Expr[Unit] = '{
+    def f(x: Int): Int = ~eval1('(f))
```

Comment:
Just a curiosity: we do eta-expansion for methods when the expected type is a function, so that a method is always applied. In this case of quotes, it seems it breaks this invariant.

Reply:
It is unrelated with the eta-expansion. This is about lifting a tree that represents a reference to a method that is outside the quote. It could not come from eta-expansion the code above could be `def f(x: Int): Int = ~eval1('{ f(1); 2 })` where `f` also needs lifting.